src/v1.js

```diff
@@ -215,6 +215,12 @@ api.declare({
   var hookId    = req.params.hookId;
   var hookDef   = req.body;
 
+  if (req.body.hookGroupId !== null && req.body.hookId !== null) {
+    if (hookGroupId !== req.body.hookGroupId || hookId !== req.body.hookId) {
+      return res.reportError('InputError', 'Hook Ids do not match', {});
+    }
+  }
+
   hookDef = _.defaults({hookGroupId, hookId}, hookDef);
```

Comment:
This line is a bit security-sensitive, since it's setting the hook's actual group and id.  And `_.defaults` has behavior that's not obvious.  Imagine this scenario:

* I am allowed to create hooks with hookGroupId "garbage" and any hookId.
* hooks in the "firefox" group have special access to important things like shipping a new version of Firefox to 500 million people
* I call `createHook('garbage', 'dustins-hook', {hookGroupId: 'firefox, task: ..})

As written, what happens?

First, the access-control check is performed against the parameters, "garbage"/"dustins-hook".  I have permission, so that check passes.  Then:
```javascript
if (req.body.hookGroupId !== undefined && req.body.hookId !== undefined) { 
```
`req.body.hookId is undefined, so this is false.  So we skip the body of that conditional.
```javascript
hookDef = _.defaults({hookGroupId, hookId}, hookDef);
```
This will set `hookDef.hookId` to `'dustins-hook'`, but it will not change `hookDef.hookGroupId`, because it's already set -- to "firefox".  This is the non-obvious behavior I mentioned.

Result: I can create a hook in the "firefox" hookGroup, that ships a copy of Firefox with malware in it.  The end!

I don't really like that scenario!  So to protect against it, step 1 is to write the conditional correctly.  But it's best to have multiple lines of defense -- what if I didn't notice that in the review?  The other thing we can do is to *always* set the hookGroupId and hookId based on the parameters -- whether they are already set or not.   So:

```javascript
hookDef.hookGroupId = hookGroupId;
hookDef.hookId = hookId;
```
This will always be correct.  If we're overwriting a value that was already in hookDef, that's OK -- we just checked that it matched anyway.